### PR TITLE
Alerting: Implement GetStatus in the remote Alertmanager struct

### PR DESCRIPTION
**Pull request by santihernandezc**
This PR implements the `GetStatus()` method in the remote Alertmanager struct.

The signature of this method has been changed for the Alertmanager interface, we now need to pass a context and return an error alongside the status.

Currently, `GetStatus` only returns the Alertmanager configuration for the internal Alertmanager. I replicated this behavior for the remote one, so instead of calling the same method on the Mimir client we retrieve the remote Alertmanager configuration and put it inside a `GettableStatus` struct.

### CODE REVIEW

File: pkg/services/ngalert/notifier/status.go [hunk 1]
```diff
@@ -1,22 +1,23 @@
...
 	config := &apimodels.PostableUserConfig{}
 	status := am.Base.GetStatus() // TODO: This should return a GettableStatus, for now it returns PostableUserConfig.
 	if status == nil {
-		return *apimodels.NewGettableStatus(&config.AlertmanagerConfig)
+		return *apimodels.NewGettableStatus(&config.AlertmanagerConfig), nil
 	}
 
 	if err := json.Unmarshal(status, config); err != nil {
-		am.logger.Error("Unable to unmarshall alertmanager config", "Err", err)
+		am.logger.Error("Unable to unmarshal alertmanager config", "err", err)
```
Comment (stevesg) [inline pkg/services/ngalert/notifier/status.go:19]: Should we break out here if that fails?
```suggestion
		am.logger.Error("Unable to unmarshal alertmanager config", "err", err)
		return nil, err
```

File: pkg/services/ngalert/remote/alertmanager.go [hunk 2]
```diff
@@ -408,8 +410,26 @@ func (am *Alertmanager) PutAlerts(ctx context.Context, alerts apimodels.Postable
...
 
-func (am *Alertmanager) GetStatus() apimodels.GettableStatus {
-	return apimodels.GettableStatus{}
+// GetStatus retrieves the remote Alertmanager configuration.
+func (am *Alertmanager) GetStatus(ctx context.Context) (apimodels.GettableStatus, error) {
+	defer func() {
+		if r := recover(); r != nil {
+			am.log.Error("Panic while getting status", "err", r)
+		}
+	}()
```
Comment (stevesg) [inline pkg/services/ngalert/remote/alertmanager.go:419]: I'm curious why we need to do this? Is there a known issue we're protecting against?

**Review by stevesg — COMMENTED** [review]
Did a first pass, looking good just a couple of questions.

File: pkg/services/ngalert/notifier/status.go (hunk 1 above)
Comment (santihernandezc) [reply pkg/services/ngalert/notifier/status.go:19]: I didn't change the behavior of this method, it's our current GetStatus in the internal Alertmanager (introduced [here](https://github.com/grafana/grafana/pull/33227/files)). I just fixed the log lines and added a context.
I agree though, not sure why we're just ignoring the error. I'll make the method return it.

File: pkg/services/ngalert/remote/alertmanager.go (hunk 2 above)
Comment (stevesg) [reply pkg/services/ngalert/remote/alertmanager.go:432]: Nasty - TIL

**Review by stevesg — APPROVED** [review]
LGTM

**Pull request merged**